analyses/oncoprint-landscape/01-plot-oncoprint.R

```diff
@@ -154,12 +144,10 @@ if (!is.null(opt$fusion_file)) {
 
 # Read in gene information from the list of genes of interest files
 if (!is.null(opt$goi_list)) {
-  goi_files <- unlist(stringr::str_split(goi_list, ",| "))
-  # Read in using the `read_genes` custom function and unlist the gene column
-  # data from the genes of interest file paths given
-  goi_list <- lapply(goi_files, read_genes)
-    # Include only the unique genes of interest
-  goi_list <- unique(unlist(goi_list))
+  # Read in using the `read_tsv()` function
+  goi_list <- readr::read_tsv(tolower(gsub(" ", "-",opt$goi_list))) %>%
+    as.matrix()
```

Comment:
I had a few questions about this step:

* Why do you need this to be a matrix? I suspect you want this to be a vector in practice?
* Why are the `tolower()` and `gsub()` steps necessary for reading in a file, where the user will specify the path to the gene list?

It's a bit tricky that the header for each GOI list is different because it encodes information about the disease type the genes are related to and one could accidentally change the file name and lose that info. I guess in the future this would be a good use case for JSON!

Reply:
> I had a few questions about this step:
> 
> * Why do you need this to be a matrix? I suspect you want this to be a vector in practice?

Good question @jaclyn-taroni, if I am not mistaken, it seems that maftools requires this to be a matrix. I did expect this to be a vector, but upon giving the `oncoplot()` function (or even the `subsetMaf()` function that was recently added) this object as a vector or even a list, I receive an error that looks like the following:

> Error in oncoplot(maf_object, clinicalFeatures = "display_group", genes = goi_list,  : 
> Oncoplot requires at-least two genes for plottng.

This error was fixed by the `as.matrix()` step.

> * Why are the `tolower()` and `gsub()` steps necessary for reading in a file, where the user will specify the path to the gene list?

This decision was made to satisfy the logic in the module shell script which uses a for loop to run `01-plot-oncoprint.R` for each broad histology and create arguments specific to each histology like the following:
> --goi_list ${oncoprint_data_directory}/"${histology}"_goi_list.tsv

Reply:
> This decision was made to satisfy the logic in the module shell script which uses a for loop to run `01-plot-oncoprint.R` for each broad histology and create arguments specific to each histology like the following:
> 
> > --goi_list ${oncoprint_data_directory}/"${histology}"_goi_list.tsv

This suggests to me that this is a problem to be solved in the shell script, not the R script.

Imagine a situation where you use `01-plot-oncoprint.R` outside the context of the shell script you are looping through and the actual file name includes capital letters - this would break!